### PR TITLE
docs: document ConsoleLogger class

### PR DESCRIPTION
**Pull request by Ryuno-Ki**
VS Code is warning me (via [CodeMetrics extension](https://marketplace.visualstudio.com/items?itemName=kisstkondoros.vscode-codemetrics)), that this class has become quite complex.

As far as I can tell, it's the only logger implementation you have in this repository. That being said, it is possible to overwrite the logger. Given that, I suggest, that you consider defining a BaseLogger class that developers can extend with their own custom logic (for example, use pino for logging).
This way, you can use the class-as-interface pattern while reducing the overhead in this class.

I was a bit unsure, which properties and methods you consider public and which private/protected. I assume, that a leading underscore indicates a „private” property.
Since the variables are pretty self-explaining, I omitted the description blocks in most of them.

Signed-off-by: André Jaenisch <andre.jaenisch@posteo.de>

### CODE REVIEW

File: src/Util/ConsoleLogger.js [hunk 1]
```diff
@@ -2,10 +2,16 @@ const chalk = require("chalk");
 const debug = require("debug")("Eleventy:Logger");
 const Readable = require("stream").Readable;
 const split = require("split");
+/** @typedef {import('stream').Readble Readable} */
```
Comment (pdehaan) [inline src/Util/ConsoleLogger.js:5]: I'm a bit confused here. Is `Readble` a typo, or am I missing something?

Comment (Ryuno-Ki) [reply src/Util/ConsoleLogger.js:5]: Obviously a typo m(

**Comment by Ryuno-Ki** [comment]
@zachleat @pdehaan What's blocking this PR from moving forward?

**Comment by pdehaan** [comment]
Looks good to me, but I don't have merge permissions on this repo so I can't land it.

**Comment by zachleat** [comment]
Will ship with 1.0.1, thanks!

**Pull request merged**